src/nix/main.cc

```diff
@@ -270,11 +270,17 @@ void mainWrapped(int argc, char * * argv)
         if (legacy) return legacy(argc, argv);
     }
 
-    verbosity = lvlNotice;
-    settings.verboseBuild = false;
     evalSettings.pureEval = true;
 
-    setLogFormat("bar");
+    if (isatty(STDERR_FILENO)) {
+        verbosity = lvlNotice;
+        settings.verboseBuild = false;
+        setLogFormat("bar");
+    } else {
+        verbosity = lvlInfo;
+        settings.verboseBuild = true;
+        setLogFormat("raw-with-logs");
+    }
```

Comment:
Printing the build logs by default is quite verbose. I think it should be opt-in (`-L`) rather than opt-out

```suggestion
    setLogFormat("raw");
    settings.verboseBuild = false;
    if (isatty(STDERR_FILENO)) {
        verbosity = lvlNotice;
    } else {
        verbosity = lvlInfo;
    }
```

Reply:
:+1: but this code disables the progress bar in all cases which is probably not what you want?

Reply:
> this code disables the progress bar in all cases which is probably not what you want?

Uh yes I meant `bar` not `raw`, thanks. Fixed